### PR TITLE
Fix gql FailureMetadata

### PR DESCRIPTION
**Pull request by smackesey**
## Summary & Motivation

Fix `failureMetadata` resolution. There was no test coverage for this.

## How I Tested These Changes

Modified existing step failure test to incorporate `failureMetadata`.

### CODE REVIEW

**Comment by smackesey** [comment]
Current dependencies on/for this PR:
* master
  * **PR #13400** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13400" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/13400?utm_source=stack-comment).

**Review by dpeng817 — APPROVED** [review]

**Review by gibsondan — APPROVED** [review]

File: python_modules/dagster-graphql/dagster_graphql_tests/graphql/test_execute_pipeline.py [hunk 1]
```diff
@@ -592,23 +601,30 @@ def test_step_failure(self, graphql_context):
...
+                "description": None,
+                "boolValue": True,
+            }
+        ]
 
-        assert len(causes) == 3
+        causes = step_run_log_entry["error"]["causes"]
+        assert len(causes) == 2
         assert [cause["message"] for cause in causes] == [
-            "Exception: Even more outer exception\n",
```
Comment (dpeng817) [inline python_modules/dagster-graphql/dagster_graphql_tests/graphql/test_execute_pipeline.py:599]: why is this being deleted / do we expect this to change?

**Review by dpeng817 — APPROVED** [review]
Seems reasonable, the fact that the `Failure` itself doesn't show up in the stack is weird but doesn't seem to have anything to do with the relevant change here.

File: python_modules/dagster-graphql/dagster_graphql_tests/graphql/test_execute_pipeline.py (hunk 1 above)
Comment (smackesey) [reply python_modules/dagster-graphql/dagster_graphql_tests/graphql/test_execute_pipeline.py:599]: special handling of `Failure`

**Comment by smackesey** [comment]
>Seems reasonable, the fact that the Failure itself doesn't show up in the stack is weird but doesn't seem to have anything to do with the relevant change here.

Pretty sure this is because of the special handling of `Failure` here, it's caught and not re-raised:

https://github.com/dagster-io/dagster/blob/148b13e72410bc37231fdb225cc5d49a02fce5a8/python_modules/dagster/dagster/_core/execution/plan/execute_plan.py#L317-L331

**Pull request merged**